modules/search-as-you-type/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -0,0 +1,432 @@
+                    parseMultiField(builder, name, parserContext, fieldName, fieldNode);
+                    iterator.remove();
+                } else if (fieldName.equals("analyzer")) {
+                    final String analyzerName = fieldNode.toString();
+                    analyzer = parserContext.getIndexAnalyzers().get(analyzerName);
+                    if (analyzer == null) {
+                        throw new MapperParsingException("Analyzer [" + analyzerName + "] not found for field  [" + name + "]");
+                    }
+                    iterator.remove();
+                } else if (fieldName.equals("max_shingle_size")) {
+                    maxShingleSize = XContentMapValues.nodeIntegerValue(fieldNode);
+                    iterator.remove();
+                }
+            }
+
+            builder.indexAnalyzer(analyzer);
+            builder.searchAnalyzer(analyzer);
+            builder.maxShingleSize(maxShingleSize);
+
+            return builder;
+        }
+    }
+
+    public static class Builder extends FieldMapper.Builder<Builder, SearchAsYouTypeFieldMapper> {
+
+        private int maxShingleSize;
+
+        public Builder(String name) {
+
+            super(name, Defaults.FIELD_TYPE, Defaults.FIELD_TYPE);
+            this.builder = this;
+        }
+
+        public Builder maxShingleSize(int maxShingleSize) {
+            if (maxShingleSize < 2) {
+                throw new MapperParsingException("[max_shingle_size] must be at least 2, got [" + maxShingleSize + "]");
+            }
+
+            if (maxShingleSize > 5) {
+                throw new MapperParsingException("[max_shingle_size] must be at most 5, got [" + maxShingleSize + "]");
+            }
+
+            this.maxShingleSize = maxShingleSize;
+
+            return builder;
+        }
+
+        @Override
+        public SearchAsYouTypeFieldType fieldType() {
+            return (SearchAsYouTypeFieldType) this.fieldType;
+        }
+
+        @Override
+        public SearchAsYouTypeFieldMapper build(BuilderContext context) {
+            setupFieldType(context);
+
+            final NamedAnalyzer indexAnalyzer = fieldType().indexAnalyzer();
+            final NamedAnalyzer searchAnalyzer = fieldType().searchAnalyzer();
+            if (indexAnalyzer.equals(searchAnalyzer) == false) {
+                throw new MapperParsingException("Index and search analyzers must be the same");
+            }
+
+            final Set<SuggesterizedFieldType> suggesterizedFieldTypes = new HashSet<>();
+
+            final SuggesterizedFieldType withEdgeNgrams = new SuggesterizedFieldType(name() + "._with_edge_ngrams");
+            final SearchAsYouTypeAnalyzer wrappedWithEdgeNGrams = SearchAsYouTypeAnalyzer.withEdgeNGrams(indexAnalyzer);
+            withEdgeNgrams.setIndexAnalyzer(new NamedAnalyzer(indexAnalyzer.name(), AnalyzerScope.INDEX, wrappedWithEdgeNGrams));
+            withEdgeNgrams.setSearchAnalyzer(indexAnalyzer);
```

Comment:
Now wondering if we should set this subfield's search analyzer to be a SearchAsYouTypeAnalyzer with both singles and edge ngrams turned off (e.g. a no-op analyzer wrapper), just so that we have the consistency of all SuggesterizedFieldTypes' analyzers being the same type

Reply:
Are you referring to the root field ?

Reply:
I was referring to the subfield without shingles and with edge ngrams, but now considering the root field too